crates/sdk/src/tx.rs

```diff
@@ -2827,6 +2827,84 @@ pub async fn build_ibc_transfer(
         masp_fee_data
     };
 
+    if let Some((target, percentage)) = &args.frontend_sus_fee {
+        match (&source, &args.ibc_shielding_data) {
+            (&MASP, None) => {
+                // NOTE: The frontend fee should NOT account for the masp fee
+                // payment amount
+                let sus_fee_amt = namada_token::Amount::from_uint(
+                    validated_amount
```

Comment:
this block of code for the percentage calculation isn't nice looking and appears in a few spots. Can we hide it in the amount module?

Reply:
Done in 3fe86a7b5e3de149a84b21ab83fa475161d237e2, but the new function does also the validation part and lives in the same module